fix: update crypto suites to require at least ECDSA w/ P-256 and SHA-256 (#295)
openid/OpenID4VC-HAIP: openid4vc-high-assurance-interoperability-profile-1_0.md

* If the credential has cryptographic holder binding, a KB-JWT, as defined in [@!I-D.ietf-oauth-sd-jwt-vc], MUST always be present when presenting an SD-JWT VC.

# Crypto Suites {#crypto-suites}

Cryptography is required by the following operations:

> - to sign and validate the signature on the Wallet or Key Attestation and their proof of possession (PoP)

Reviewer: Also, the sentences below make it clear which signature we're validating (e.g. Issuer's signature) while this just uses "signature"

- to sign and validate the signature of the status information of the Verifiable Credential or Wallet Attestation
- to sign and validate the Issuer's signature on the Verifiable Credential
- to sign and validate the Holder's signature on the Verifiable Presentation
- to sign and validate the Verifier's signature on the Presentation Request

> All parties MUST, at a minimum, support ECDSA with P-256 and SHA-256 (JOSE algorithm identifier `ES256`; COSE algorithm identifier `-7`, as applicable) for the following:

Reviewer: "All parties" is less specific than "Issuers, Holders, and Verifiers". Why not keep it?

```suggestion
Issuers, Verifiers, and Wallets MUST, at a minimum, support ECDSA with P-256 and SHA-256 (JOSE algorithm identifier `ES256`; COSE algorithm identifier `-7`, as applicable) for the following purposes:
```

- Issuers
  - validating Wallet Attestations (including PoP) when Annex E of [@!OIDF.OID4VCI] is used;
  - validating Key Attestations when Annex D of [@!OIDF.OID4VCI] is used.
- Verifiers
  - validating the signature of the Verifiable Presentation, e.g., KB-JWT of an SD-JWT VC, or `deviceSignature` CBOR structure in case of ISO mdocs. Verifiers are assumed to determine in advance the cryptographic suites supported by the ecosystem, e.g. mDL Issuers/Verifiers implementing ISO mdocs.
- Wallets
  - validating signed presentation requests.

> Although support for the above algorithm is mandatory, an Issuer, Verifier, or Wallet MAY reject a particular cryptographic suite or the corresponding protected artifact if there are reasonable security, privacy, or policy concerns (e.g., suspected compromise or non-compliance). Such decisions are out of scope of the algorithm requirements but can be necessary in real-world deployments.

Reviewer: Not sure I understand this. What do you mean with

> MAY reject a particular cryptographic suite or the corresponding protected artifact

Can they refuse the use of ECDSA with P-256 and SHA-256?

Reviewer 2: If a party refuses to use an algorithm, are there a mechanism to notify the other party of a reason and supported alternatives?

> Ecosystem-specific profiles MAY mandate additional or alternative cryptographic suites.

Reviewer 3: Discussed on this morning's WG call:

"alternative" here seems weird given line 290 has a MUST requirement to support P-256?

* If the credential has cryptographic holder binding, a KB-JWT, as defined in [@!I-D.ietf-oauth-sd-jwt-vc], MUST always be present when presenting an SD-JWT VC.

# Crypto Suites {#crypto-suites}

> Issuers, Verifiers, and Wallets MUST, at a minimum, support ECDSA with P-256 and SHA-256 (JOSE algorithm identifier `ES256`; COSE algorithm identifier `-7`, as applicable) for the following purposes:

Reviewer 4: ```suggestion
Issuers, Verifiers, and Wallets MUST, at a minimum, support ECDSA with P-256 and SHA-256 (JOSE algorithm identifier `ES256`; COSE algorithm identifier `-7`, as applicable) for the purpose of validating the following:
```

* If the credential has cryptographic holder binding, a KB-JWT, as defined in [@!I-D.ietf-oauth-sd-jwt-vc], MUST always be present when presenting an SD-JWT VC.

# Crypto Suites {#crypto-suites}

> Cryptography is required by the following operations:
> 

Author: Perhaps we can just remove this, since we are repeating this below.
```suggestion
```

> - to sign and validate the signature on the Wallet or Key Attestation, and their proof of possession (PoP)
> - to sign and validate the signature of the status information of the Verifiable Credential or Wallet Attestation
> - to sign and validate the Issuer's signature on the Verifiable Credential
> - to sign and validate the Holder's signature on the Verifiable Presentation
> - to sign and validate the Verifier's signature on the Presentation Request

Author: Perhaps we can just remove this, since we are repeating this below.
```suggestion
```

Reviewer 4: agreed.

All parties MUST, at a minimum, support ECDSA with P-256 and SHA-256 (JOSE algorithm identifier `ES256`; COSE algorithm identifier `-7`, as applicable) for the following:

- Issuers
  - validating Wallet Attestations (including PoP) when Annex E of [@!OIDF.OID4VCI] is used;
  - validating Key Attestations when Annex D of [@!OIDF.OID4VCI] is used.
- Verifiers

>   - validating the signature of the Verifiable Presentation, e.g., KB-JWT of an SD-JWT VC, or `deviceSignature` CBOR structure in case of ISO mdocs. Verifiers are assumed to determine in advance the cryptographic suites supported by the ecosystem, e.g. mDL Issuers/Verifiers implementing ISO mdocs.

Author: ```suggestion
  - validating the signature of the Verifiable Presentation, e.g., KB-JWT of an SD-JWT VC, or `deviceSignature` CBOR structure in case of ISO mdocs. Verifiers are assumed to determine in advance the cryptographic suites supported by the ecosystem, e.g. mDL Issuers/Verifiers implementing ISO mdocs.
  - validating the status information of the Verifiable Credential or Wallet Attestation.
```
